### PR TITLE
Add API Usage by Application implementation.

### PR DESCRIPTION
**Pull request by menakaj**
## Purpose
Add the api usage by application implementation. 

Fixes https://github.com/wso2/product-apim/issues/3783

## Approach

The API Usage By Applications UI

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/stats/api-usage-by-applications/js/stats.js [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                } else {
+                    jagg.message({content: json.message, type: "error"});
+                }
+            }
+        }, "json");
+}
+
+function getDateTime(currentDay, fromDay) {
+    to = convertTimeString(currentDay);
+    from = convertTimeString(fromDay);
```
Comment (chamilaadhi) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/stats/api-usage-by-applications/js/stats.js]: I don't think this is implemented to work with different timezones (browser in a different timezone and the servers are in a different one). Please check this. See https://github.com/wso2/carbon-apimgt/pull/5758/files#diff-c6b01b5ecf5cd966a32cf31947e85592R148 a sample code

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/stats/api-usage-by-applications/template.jag [hunk 2]
```diff
@@ -0,0 +1,74 @@
+<% jagg.template("stats/api-usage-by-applications", function(inputs, outputs, jagg) { %>
+<div class="page-header"> <h2><%=i18n.localize("API Usage By Applications")%></h2></div>
+<div id="apiUsageByApplication-stat-page" class="stat-page">
+    <%
+        //show content only if analytics enabled
+        if(org.wso2.carbon.apimgt.impl.utils.APIUtil.isAnalyticsEnabled()) {
```
Comment (sameeragunarathne) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/stats/api-usage-by-applications/template.jag]: formatting issue

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/stats/api-usage-destination/js/stats.js [hunk 3]
```diff
@@ -141,7 +141,6 @@ var drawAPIUsageByDestination = function(from,to){
                              $('#destinationBasedUsageTable_paginate').show();
                      },
                     });
-                    $('select').css('width','60px');
```
Comment (fazlan-nazeem) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/stats/api-usage-destination/js/stats.js]: why is this removed?

Comment (menakaj) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/stats/api-usage-destination/js/stats.js]: This overrides the default wso2 styling and causes API Provider drop-down list to shrink.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/stats/api-usage-by-applications/template.jag (hunk 2 above)
Comment (menakaj) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/stats/api-usage-by-applications/template.jag]: Fixed https://github.com/wso2/carbon-apimgt/pull/5897/commits/13c7c0005855503a4d68ae06df35fec273217113
